File: youzi_v2/frontend/src/api/channels.ts
import { api } from './client'

export interface Channel {
  code: string
  nameZh: string
  nameEn: string
  country: string
  category: string
  note: string
  sortOrder: number
  isActive: boolean
  createdTime: string
  updatedTime: string
}

export interface ChannelListResponse {
  items: Channel[]
  total: number
  limit: number
  offset: number
}

export interface ChannelPayload {
  code: string
  nameZh: string
  nameEn?: string
  country: string
  category: string
  note?: string
  sortOrder?: number
  isActive?: boolean
}

export interface ChannelSeedResult {
  inserted: number
  updated: number
  total: number
}

export async function getChannelMeta(): Promise<{ categories: string[] }> {
  return api<{ categories: string[] }>('/api/v1/channels/meta')
}

export async function listChannels(params?: {
  search?: string
  country?: string
  category?: string
  activeOnly?: boolean
  limit?: number
  offset?: number
}): Promise<ChannelListResponse> {
  const q: Record<string, string | number | boolean> = {}
  if (params?.search) q.search = params.search
  if (params?.country) q.country = params.country
  if (params?.category) q.category = params.category
  if (params?.activeOnly !== undefined) q.activeOnly = params.activeOnly
  if (params?.limit !== undefined) q.limit = params.limit
  if (params?.offset !== undefined) q.offset = params.offset
  return api<ChannelListResponse>('/api/v1/channels', { query: q })
}

export async function seedDefaultChannels(): Promise<ChannelSeedResult> {
  return api<ChannelSeedResult>('/api/v1/channels/seed-defaults', { method: 'POST' })
}

export async function createChannel(payload: ChannelPayload): Promise<Channel> {
  return api<Channel>('/api/v1/channels', {
    method: 'POST',
    body: payload,
  })
}

export async function updateChannel(
  code: string,
  payload: Partial<Omit<ChannelPayload, 'code'>>,
): Promise<Channel> {
  return api<Channel>(`/api/v1/channels/${encodeURIComponent(code)}`, {
    method: 'PATCH',
    body: payload,
  })
}

export async function deleteChannel(code: string): Promise<void> {
  await api(`/api/v1/channels/${encodeURIComponent(code)}`, { method: 'DELETE' })
}

File: youzi_v2/schemas/channels.py
from pydantic import AliasChoices, BaseModel, ConfigDict, Field


class ChannelIn(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    code: str = Field(min_length=1)
    name_zh: str = Field(
        default="",
        validation_alias=AliasChoices("nameZh", "name_zh"),
    )
    name_en: str = Field(
        default="",
        validation_alias=AliasChoices("nameEn", "name_en"),
    )
    country: str = ""
    category: str = ""
    note: str = ""
    sort_order: int = Field(default=0, validation_alias=AliasChoices("sortOrder", "sort_order"))
    is_active: bool = Field(default=True, validation_alias=AliasChoices("isActive", "is_active"))


class ChannelUpdateIn(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    code: str | None = Field(default=None, min_length=1)
    name_zh: str | None = Field(default=None, validation_alias=AliasChoices("nameZh", "name_zh"))
    name_en: str | None = Field(default=None, validation_alias=AliasChoices("nameEn", "name_en"))
    country: str | None = None
    category: str | None = None
    note: str | None = None
    sort_order: int | None = Field(default=None, validation_alias=AliasChoices("sortOrder", "sort_order"))
    is_active: bool | None = Field(default=None, validation_alias=AliasChoices("isActive", "is_active"))


class ChannelSeedResult(BaseModel):
    model_config = ConfigDict(populate_by_name=True)

    inserted: int
    updated: int
    total: int
